CONTRIBUTING.md

```diff
@@ -1,27 +1,92 @@
+8. [OPTIONAL] You may run into unnecessary commits, such as merge conflicts or minor refactorings. In that case, always `git rebase -i`  and [squash](https://www.git-tower.com/learn/git/faq/git-squash) all your commits to one commit. 
+9. Please do not merge master into your branch as you develop your pull request; instead, rebase your branch on top of the latest master if your pull request branch is long-lived.
+    1. How to regularly update your PR with master:
+        
+        ```
+        git checkout main
+        git pull
+        git checkout my/branch
+        git merge main
+        git rebase -i // squash commits!
+        git push
+        ```
+        
+
+### Submitting Bug Issues
+
+1. If you identified a bug, please create a new issue or discuss it in our forums. 
+2. In the description, provide the following:
+    - What command(s) is the bug in?
+    - Describe the bug
+    - Concrete steps to reproduce the bug
+
+See [this guide](https://stackoverflow.com/help/mcve) on how to create a minimal, complete, and verifiable example.
+
+### Submitting Feature Issues
+
+1. If you identified a feature, please create a new issue or discuss it in our forums. 
+2. In the description, provide the following:
+    - Describe the feature you would like
+    - Additional context
+
+### Adding Tests
+
+If the change being proposed alters code, it is either adding new functionality to zkevm-circuits, or fixing existing, broken functionality. In both of these cases, the pull request should include one or more tests to ensure that zkevm-circuits does not regress in the future.
+
+Types of tests include:
+
+- **Unit tests**: Functions which have very specific tasks should be unit tested.
+- **Integration tests**: For general purpose, far reaching functionality, integration tests should be added. The best way to add a new integration test is to look at existing ones and follow the style.
 
 ## General considerations
-- Make atomic PRs. Focused on a single feature or topic. Don't mix different topics or issues in the same PR if it can be avoided.
-- For PRs that change significantly the codebase or the way on which some procedure is done (test-suite, cross-crate structures, module refactors) please file an issue and wait until this feature/proposal gets approved by the repo maintainers.
-- Don't set PRs as ready for review until they're not completed from the dev perspective and also are up-to-date with `main`/`master`. Also, if any changes are requested, bring back the PR to `Draft` until it's ready again with the aforementioned requirements.
+
+- Don't set PRs as ready for review until they're not completed from the dev perspective and also are up-to-date with `main`/`master`. Also, if any changes are requested, bring back the PR to `Draft` until it's ready again with the aforementioned requirements.
```

Comment:
> Also, if any changes are requested, bring back the PR to `Draft` until it's ready again with the aforementioned requirements.

We're not usually doing this.  I think having unresolved requests from a reviewer, or having the PR marked as changes requested is enough to signal that the PR is not ready to be merged and the PR submitter has work to do.